File: 3249-minimum-number-of-operations-to-make-array-xor-equal-to-k/minimum-number-of-operations-to-make-array-xor-equal-to-k.dart
class Solution {
  int minOperations(List<int> nums, int k) {
    int xorRes = k,res = 0;
    for (int n in nums) xorRes ^= n;
    for(;xorRes > 0; xorRes >>= 1) if (xorRes & 1 == 1) res++;
    return res;
  }
}

File: 3249-minimum-number-of-operations-to-make-array-xor-equal-to-k/minimum-number-of-operations-to-make-array-xor-equal-to-k.rs
impl Solution {
    pub fn min_operations(nums: Vec<i32>, k: i32) -> i32 {
        let mut xor_res: i32 = k;
        let mut res: i32 = 0;
        for n in nums {
            xor_res ^= n;
        }
        while xor_res > 0 {
            if xor_res & 1 == 1 {
                res += 1;
            }
            xor_res >>= 1;
        }
        res
    }
}